### PR TITLE
send options ping

### PR DESCRIPTION
**Pull request by xquanluu**

### CODE REVIEW

**Review by davehorton — CHANGES_REQUESTED** [review]
I think we need to revise this.  Unlike outbound REGISTERs, where we only send from one SBC, we should send OPTIONS ping from all SBCs.  Thus the functionality should be factored out of sip-trunk-register.js into its own file.

**Comment by xquanluu** [comment]
Fixed

File: lib/sip-trunk-options-ping.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    }
+  }
+}
+
+module.exports = async(logger, srf) => {
+  const updateSipGatewayOptsBot = async(logger, srf) => {
+    try {
+
+      const { lookupSipGatewaysByFilters } = srf.locals.dbHelpers;
+      const gws = await lookupSipGatewaysByFilters({send_options_ping: true, outbound: true, is_active: true});
```
Comment (davehorton) [inline lib/sip-trunk-options-ping.js:66]: is_active: true means that once we mark a gateway as inactive/down we will never test it again..hmmm

Comment (davehorton) [reply lib/sip-trunk-options-ping.js:66]: any ideas on this?  Do we in fact need a new property in the sip_gateways table?  Or should we keep track of all active/down gateways in redis?

**Review by davehorton — CHANGES_REQUESTED** [review]

File: lib/sip-trunk-options-ping.js (hunk 1 above)
Comment (davehorton) [reply lib/sip-trunk-options-ping.js:66]: I guess we really have two different pieces of information we are tracking:
- active/inactive -- do we want to use this gateway at all
- up/down -- for active gateways, is it responding to options pings

I kind of like not changing the mysql schema and instead keeping a list in redis of all active/down gateways

**Comment by davehorton** [comment]
@xquanluu we're almost ready to merge this, as the first feature of 0.8.6.  Can you rebase

File: lib/sip-trunk-options-ping.js (hunk 1 above)
Comment (xquanluu) [reply lib/sip-trunk-options-ping.js:69]: Fixed

**Review by davehorton — APPROVED** [review]

**Pull request merged**